da/avail/avail.go

```diff
@@ -0,0 +1,187 @@
+
+	return nil
+}
+
+// SubmitBlock submits a block to DA layer.
+func (c *DataAvailabilityLayerClient) SubmitBlock(ctx context.Context, block *types.Block) da.ResultSubmitBlock {
+
+	data, err := block.MarshalBinary()
+	if err != nil {
+		return da.ResultSubmitBlock{
+			BaseResult: da.BaseResult{
+				Code:    da.StatusError,
+				Message: err.Error(),
+			},
+		}
+	}
+
+	txHash, err := datasubmit.SubmitData(c.config.Size, c.config.ApiURL, c.config.Seed, c.config.AppID, data)
+
+	if err != nil {
+		return da.ResultSubmitBlock{
+			BaseResult: da.BaseResult{
+				Code:    da.StatusError,
+				Message: err.Error(),
+			},
+		}
+	}
+
+	return da.ResultSubmitBlock{
+		BaseResult: da.BaseResult{
+			Code:     da.StatusSuccess,
+			Message:  "tx hash: " + hex.EncodeToString(txHash[:]),
+			DAHeight: 1,
+		},
+	}
+}
+
+// CheckBlockAvailability queries DA layer to check data availability of block.
+func (c *DataAvailabilityLayerClient) CheckBlockAvailability(ctx context.Context, dataLayerHeight uint64) da.ResultCheckBlock {
+
+	blockNumber := dataLayerHeight
+	confidenceURL := fmt.Sprintf(c.config.BaseURL+"/confidence/%d", blockNumber)
+
+	response, err := http.Get(confidenceURL)
+
+	if err != nil {
+		return da.ResultCheckBlock{
+			BaseResult: da.BaseResult{
+				Code:    da.StatusError,
+				Message: err.Error(),
+			},
+		}
+	}
+
+	responseData, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return da.ResultCheckBlock{
+			BaseResult: da.BaseResult{
+				Code:    da.StatusError,
+				Message: err.Error(),
+			},
+		}
+	}
+
+	var confidenceObject Confidence
+	json.Unmarshal(responseData, &confidenceObject)
+
+	return da.ResultCheckBlock{
+		BaseResult: da.BaseResult{
+			Code:     da.StatusSuccess,
+			DAHeight: uint64(confidenceObject.Block),
+		},
+		DataAvailable: confidenceObject.Confidence > float64(c.config.confidence),
+	}
+}
+
+//RetrieveBlocks gets the block from DA layer.
+
+func (c *DataAvailabilityLayerClient) RetrieveBlocks(ctx context.Context, dataLayerHeight uint64) da.ResultRetrieveBlocks {
+
+	blocks := make([]*types.Block, 1)
+	blocks[0] = new(types.Block)
+
+	blockNumber := dataLayerHeight
+	appDataURL := fmt.Sprintf(c.config.BaseURL+"/appdata/%d?decode=true", blockNumber)
+	response, err := http.Get(appDataURL)
+	if err != nil {
+		return da.ResultRetrieveBlocks{
+			BaseResult: da.BaseResult{
+				Code:    da.StatusError,
+				Message: err.Error(),
+			},
+		}
+	}
+	responseData, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return da.ResultRetrieveBlocks{
+			BaseResult: da.BaseResult{
+				Code:    da.StatusError,
+				Message: err.Error(),
+			},
+		}
+	}
+	var appDataObject AppData
+	json.Unmarshal(responseData, &appDataObject)
+
+	return da.ResultRetrieveBlocks{
+		BaseResult: da.BaseResult{
+			Code:     da.StatusSuccess,
+			DAHeight: uint64(appDataObject.Block),
+			Message:  "block data: " + appDataObject.Extrinsics,
+		},
+		Blocks: blocks,
```

Comment:
Shoudn't we do here something like `Blocks: []types.Block{appDataObject}`, probably creating block from appdata object previously? And remove block variable.

Reply:
@aterentic-ethernal 
types.Block and appDataObject are two different types. I don't think we can do that. Please share more details on how to do it.

Reply:
It looks like `blocks` is always empty, and it doesn't contain extrinsics, I'm not sure thats intended behavior.

Reply:
Yeah but we cant able to add extrinsics to block as their types are different .Is there any alternative to do that?Please throw some light on this

Reply:
Each extrinsic is base64 encoded transaction, we can decode them and create `Block/Data/Txs`. But I'm not sure how to create `IntermediateStateRoots` and `SignedHeader` which is also required to create a `Block`.

Reply:
@aterentic-ethernal , I have assigned the extrinsics to the byte array of txs and taken intermediate roots as nil,please look into it and correct me If I did anything wrong

Reply:
Gentle ping @aterentic-ethernal

Reply:
Sorry, my notification mail was lost somewhere in the mess, Ill review all changes asap :)

Reply:
Well I'm not sure how rollkit utilizes `Blocks` field of the result, but from the code, it looks like that field contains one empty block. I assume there should be data retreived from avail. Not sure also what is purpose of the message. Am I missing something?